### PR TITLE
Tangible Total Validation

### PR DESCRIPTION
**Pull request by BSaunders95**
Add tangible assets note total validation.

Resolves SFA-986

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java [hunk 1]
```diff
@@ -0,0 +1,931 @@
...
+    private static final String COST_AT_PERIOD_END = ".cost.at_period_end";
+    private static final String DEPRECIATION_AT_PERIOD_START = ".depreciation.at_period_start";
+    private static final String CHARGE_FOR_YEAR = ".depreciation.charge_for_year";
+    private static final String ON_DISPOSALS = ".depreciation.on_disposals";
+    private static final String OTHER_ADJUSTMENTS = ".depreciation.other_adjustments";
+    private static final String DEPRECIATION_AT_PERIOD_END = ".depreciation.at_period_end";
+    private static final String NET_BOOK_VALUE_CURRENT_PERIOD = ".net_book_value_at_end_of_current_period";
+    private static final String NET_BOOK_VALUE_PREVIOUS_PERIOD = ".net_book_value_at_end_of_previous_period";
+
+    public Errors validateTangibleAssets(TangibleAssets tangibleAssets, Transaction transaction, String companyAccountsId, HttpServletRequest request)
```
Comment (caismith-ch) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java]: Doesn't look like companyAccountsId and request are ever used.

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java [hunk 2]
```diff
@@ -0,0 +1,931 @@
...
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getCost() == null || tangibleAssetsResource.getCost().getAtPeriodEnd() == null) {
+
+                addError(errors, valueRequired, getJsonPath(subResource, COST_AT_PERIOD_END));
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getDepreciation() != null) {
```
Comment (leahtaylor) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java]: Not 100% sure on the validation logic require - but could we merge these two if statements to be 

` if ((tangibleAssetsResource.getDepreciation() != null) && (hasDepreciationFieldsSet(tangibleAssetsResource))) {
`

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java (hunk 1 above)
Comment (caismith-ch) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java]: Spoke to @BSaunders95 - these will be required for future so will remain.

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java (hunk 2 above)
Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java]: Multiple if statements have been compounded into 1

**Pull request merged**